File: Examples/test-suite/ocaml/cpp_nodefault_runme.ml
open Swig
open Cpp_nodefault

let foo1 = new_Foo '(1, 2)
let _ = foo1 -> "[a]" (5)
let _ = assert ((foo1 -> "[a]" () as int) = 5)

let foo2 = _create '(1, 2)
let _ = _consume '(foo1,foo2)

let bar1 = new_Bar '()
let gvar = _gvar '()
let args = (C_list [ gvar ; foo2 ])
let _ = bar1 -> "consume" (args)
let foo3 = bar1 -> "create" (1, 2)
let _ = foo3 -> "[a]" (6)
let _ = assert ((foo3 -> "[a]" () as int) = 6)

File: Examples/test-suite/lua/cpp_nodefault_runme.lua
require("import")
require("cpp_nodefault")
cn=cpp_nodefault
catch_undef_globs() -- catch "undefined" global variables

foo1 = cn.Foo(1,2)
foo1.a = 5
assert(foo1.a == 5)

foo2 = cn.create(1,2)

cn.consume(foo1,foo2)

bar1 = cn.Bar()
bar1:consume(cn.gvar, foo2)
foo3 = bar1:create(1,2)

foo3.a = 6
assert(foo3.a == 6)
